sdks/go/pkg/beam/core/runtime/exec/datasource.go

```diff
@@ -174,6 +174,7 @@ func (n *DataSource) Process(ctx context.Context) error {
 func (n *DataSource) FinishBundle(ctx context.Context) error {
 	log.Infof(ctx, "DataSource: %d elements in %d ns", atomic.LoadInt64(&n.count), time.Now().Sub(n.start))
 	n.source = nil
+	atomic.StoreInt64(&n.count, 0)
```

Comment:
I mispoke,please move this to after the finishBundle call, but before returning the error.
The following should be correct, and reduce the number of 

err :=  n.Out.FinishBundle(ctx)
atomic.StoreInt64(&n.count, 0)
return err